block-layout/server.py

```diff
@@ -147,6 +147,20 @@ def node_name(node):
         print("Attempting to make a name for node %s [using uri] "%(repr(node)))
         return node['uri']
 
+@app.route('/nodes/type/<node_type>')
+def nodes_type(node_type):
+    '''Return all the nodes with the type node_type'''
+
+    node_type = urllib.parse.unquote(node_type)
+    result =  {}
+
+    q = """MATCH (n:%s) RETURN n""" % node_type
```

Comment:
I'm not sure about pasting node_type directly into the query string here. It looks a bit like the Cypher equivalent of an SQL injection attack. urllib.parse.unquote won't do much to stop it. It might be safe but I need to do some more checking to verify it.